Subject: [PATCH] Release the wakelock before worker thread termination

This case may happen when a new weather update is immediately scheduled
(for example after the location has changed), while another request is
in progress, and network is not available.

diff --git a/src/com/cyanogenmod/lockclock/weather/WeatherUpdateService.java b/src/com/cyanogenmod/lockclock/weather/WeatherUpdateService.java
index 61b7aa0..5ad3a8f 100644
--- a/src/com/cyanogenmod/lockclock/weather/WeatherUpdateService.java
+++ b/src/com/cyanogenmod/lockclock/weather/WeatherUpdateService.java
@@ -313,7 +313,11 @@
 
         public void tearDown() {
             if (D) Log.d(TAG, "Tearing down worker thread");
-            if (isProcessing()) mWeatherManager.cancelRequest(mRequestId);
+            if (isProcessing()) {
+                cancelTimeoutAlarm();
+                mWeatherManager.cancelRequest(mRequestId);
+                broadcastAndCleanUp(true);
+            }
             quit();
         }
 
